sql/core/src/test/scala/org/apache/spark/sql/DataFrameRangeSuite.scala

```diff
@@ -127,4 +133,28 @@ class DataFrameRangeSuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("Cancelling stage in a query with Range.") {
+    val listener = new SparkListener {
+      override def onTaskStart(taskStart: SparkListenerTaskStart): Unit = {
+        Thread.sleep(100)
+        sparkContext.cancelStage(taskStart.stageId)
+      }
+    }
+
+    sparkContext.addSparkListener(listener)
+    for (codegen <- Seq(true, false)) {
+      withSQLConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> codegen.toString()) {
+        val ex = intercept[SparkException] {
+          spark.range(100000L).crossJoin(spark.range(100000L))
+            .toDF("a", "b").agg(sum("a"), sum("b")).collect()
+        }
+        val msg = if (ex.getCause() != null) ex.getCause().getMessage() else ex.getMessage()
```

Comment:
Either this:
`org.apache.spark.SparkException: Job 43 cancelled because Stage 85 was cancelled`
or the same wrapped by this:
https://github.com/apache/spark/blob/fb3081d3b38a50aa5e023c603e1b191e57f7c876/core/src/main/scala/org/apache/spark/util/ThreadUtils.scala#L205